### PR TITLE
[refactor](datalake) return the error status instead of static_cast<void>

### PR DESCRIPTION
**Pull request by AshinGau**
## Proposed changes

Followup #34797
`static_cast<void>` has ignored the wrong status, some of them should make the query finished with error status, so replace `static_cast<void>`  with `RETURN_IF_ERROR`.

### Remaining Works
The following three scenarios need to be handled separately and cannot be simply replaced:
1. The outer function returns void;
2. Call status function inner constructors or destructors;
3. Call status function with best effort, and should ignore the wrong status.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by AshinGau** [comment]
run buildall

File: be/src/vec/exec/format/table/hudi_jni_reader.h [hunk 1]
```diff
@@ -63,6 +63,13 @@ class HudiJniReader : public GenericReader {
     Status init_reader(
             std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range);
 
+    Status close() override {
```
Comment (morningman) [inline be/src/vec/exec/format/table/hudi_jni_reader.h:66]: Same as parent, no need to override

File: be/src/vec/exec/format/avro/avro_jni_reader.h [hunk 2]
```diff
@@ -85,6 +85,13 @@ class AvroJNIReader : public GenericReader {
 
     TypeDescriptor convert_complex_type(const rapidjson::Document::ConstObject child_schema);
 
+    Status close() override {
```
Comment (morningman) [inline be/src/vec/exec/format/avro/avro_jni_reader.h:88]: Same as parent, no need to override

File: be/src/vec/exec/format/table/max_compute_jni_reader.h [hunk 3]
```diff
@@ -68,6 +68,13 @@ class MaxComputeJniReader : public GenericReader {
     Status init_reader(
             std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range);
 
+    Status close() override {
```
Comment (morningman) [inline be/src/vec/exec/format/table/max_compute_jni_reader.h:71]: Same as parent, no need to override

File: be/src/vec/exec/format/table/paimon_reader.h [hunk 4]
```diff
@@ -66,6 +66,13 @@ class PaimonJniReader : public GenericReader {
     Status init_reader(
             std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range);
 
+    Status close() override {
```
Comment (morningman) [inline be/src/vec/exec/format/table/paimon_reader.h:69]: Same as parent, no need to override

File: be/src/vec/exec/format/table/trino_connector_jni_reader.h [hunk 5]
```diff
@@ -61,6 +61,13 @@ class TrinoConnectorJniReader : public GenericReader {
     Status init_reader(
             std::unordered_map<std::string, ColumnValueRangeType>* colname_to_value_range);
 
+    Status close() override {
```
Comment (morningman) [inline be/src/vec/exec/format/table/trino_connector_jni_reader.h:64]: ditto

File: be/src/olap/wal/wal_reader.cpp [hunk 6]
```diff
@@ -50,11 +50,8 @@ Status WalReader::init() {
 }
 
 Status WalReader::finalize() {
-    if (file_reader != nullptr) {
-        auto st = file_reader->close();
-        if (!st.ok()) {
-            LOG(WARNING) << "fail to close wal " << _file_name << " st= " << st.to_string();
-        }
+    if (file_reader) {
+        return file_reader->close();
```
[the remark this answers is not in the document]
Comment (w41ter) [reply be/src/olap/wal/wal_reader.cpp:54]: It should have been returned but was left out. This change is LGTM.

File: be/src/vec/exec/format/wal/wal_reader.cpp [hunk 7]
```diff
@@ -30,12 +30,6 @@ WalReader::WalReader(RuntimeState* state) : _state(state) {
     _wal_id = state->wal_id();
 }
 
-WalReader::~WalReader() {
```
[the remark this answers is not in the document]
Comment (w41ter) [reply be/src/vec/exec/format/wal/wal_reader.cpp:33]: LGTM

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run buildall

**Comment by AshinGau** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by w41ter — APPROVED** [review]

**Pull request merged**